src/format/Kdbx3Reader.h

```diff
@@ -0,0 +1,62 @@
+/*
+ *  Copyright (C) 2010 Felix Geyer <debfx@fobos.de>
+ *
+ *  This program is free software: you can redistribute it and/or modify
+ *  it under the terms of the GNU General Public License as published by
+ *  the Free Software Foundation, either version 2 or (at your option)
+ *  version 3 of the License.
+ *
+ *  This program is distributed in the hope that it will be useful,
+ *  but WITHOUT ANY WARRANTY; without even the implied warranty of
+ *  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ *  GNU General Public License for more details.
+ *
+ *  You should have received a copy of the GNU General Public License
+ *  along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+
+#ifndef KEEPASSX_KDBX3READER_H
+#define KEEPASSX_KDBX3READER_H
+
+#include <QCoreApplication>
+
+#include "format/KeePass2Reader.h"
+#include "keys/CompositeKey.h"
+
+class Database;
+class QIODevice;
+
+class Kdbx3Reader : public BaseKeePass2Reader
+{
+    Q_DECLARE_TR_FUNCTIONS(Kdbx3Reader)
+
+public:
+    Kdbx3Reader();
+
+    using BaseKeePass2Reader::readDatabase;
```

Comment:
This is needed because `readDatabase` is overloaded and the other overload is not virtual.

If this is omitted, the other overload will be hidden.

Reply:
That is not true, they are both declared virtual and have different signatures there wouldn't be any shadowing: https://github.com/keepassxreboot/keepassxc/pull/1179/files/f3d019edd927984f249fb57caf0ca35903f5a4b4#diff-3e756dd62a4ab53a029aec11b703b700R38

Reply:
Problem is that C++ is a little wonky when it comes to overloads and overrides. When you override a method it will shadow all overloads of the parent class. So this is indeed needed here.